tests/snapshots/test_snapshot_worker.py

```diff
@@ -45,44 +106,35 @@
     "first_release_id": None,
 }
 
+test_data = [
+    (90, None),
+    (100, None),
+    (110, None),
+    (120, InsertBatch([PROCESSED])),
+    (210, InsertBatch([PROCESSED])),
+]
 
-class TestSnapshotWorker:
-
-    test_data = [
-        (INSERT_MSG % {"xid": 90}, None),
-        (INSERT_MSG % {"xid": 100}, None),
-        (INSERT_MSG % {"xid": 110}, None),
-        (INSERT_MSG % {"xid": 120}, InsertBatch([PROCESSED])),
-        (INSERT_MSG % {"xid": 210}, InsertBatch([PROCESSED])),
-    ]
 
-    @pytest.mark.parametrize("value, expected", test_data)
-    def test_send_message(
-        self, value: str, expected: Optional[ProcessedMessage],
-    ) -> None:
-        storage = get_storage(StorageKey.GROUPEDMESSAGES)
-        snapshot_id = uuid1()
-        transact_data = TransactionData(xmin=100, xmax=200, xip_list=[120, 130])
+@pytest.mark.parametrize("xid, expected", test_data)
+def test_send_message(xid: int, expected: Optional[ProcessedMessage]) -> None:
+    processor = (
+        get_writable_storage(StorageKey.GROUPEDMESSAGES)
+        .get_table_writer()
+        .get_stream_loader()
+        .get_processor()
+    )
 
-        worker = SnapshotAwareWorker(
-            storage=storage,
-            producer=FakeConfluentKafkaProducer(),
-            snapshot_id=str(snapshot_id),
-            transaction_data=transact_data,
-            replacements_topic=None,
-            metrics=DummyMetricsBackend(strict=True),
-        )
+    worker = SnapshotProcessor(
+        processor=processor,
+        snapshot_id=SnapshotId(str(uuid1())),
+        transaction_data=TransactionData(
+            xmin=Xid(100), xmax=Xid(200), xip_list=[Xid(120), Xid(130)]
+        ),
+    )
 
-        message: Message[KafkaPayload] = Message(
-            Partition(Topic("topic"), 0),
-            1,
-            KafkaPayload(
-                None,
-                value.encode("utf-8"),
-                [("table", "sentry_groupedmessage".encode())],
-            ),
-            datetime.now(),
-        )
+    ret = worker.process_message(
+        get_insert_event(xid),
+        KafkaMessageMetadata(offset=1, partition=0, timestamp=datetime.now()),
+    )
 
-        ret = worker.process_message(message)
-        assert ret == expected
+    assert ret == expected
```

Comment:
This is far too coupled to the real world implementation of the grouped message processor right now in my opinion but I didn't want to change the entire test.